packages/dom-adapters/src/BlockToolAdapter/index.ts

```diff
@@ -189,7 +190,7 @@ export class BlockToolAdapter {
 
         range.insertNode(textNode);
 
-        caretAdapter.updateIndex([start + text.length, start + text.length]);
+        this.#caretAdapter.updateIndex([[start + text.length, start + text.length], composeDataIndex(key), this.#blockIndex]);
```

Comment:
its hard to understand when all params passed as array. I'd suggest creating private (or utility) method like `composeInputCaretIndex(start, end, key, blockIndex)`